engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -0,0 +1,115 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedObjectTypingResult, TypingResult}
+import cats.instances.list._
+
+import scala.collection.JavaConverters._
+import pl.touk.nussknacker.engine.util.Implicits._
+import pl.touk.nussknacker.engine.util.validated.ValidatedSyntax
+
+object aggregates {
+
+  private val syntax = ValidatedSyntax[String]
+  import syntax._
+
+  object MaxAggregator extends ReducingAggregator {
+
+    override type Element = Number
+
+    override def zero: Number = Double.MinValue
+
+    override def add: (Number, Number) => Number = (n1, n2) => Math.max(n1.doubleValue(), n2.doubleValue())
+
+    override def zeroType: TypingResult = Typed[Number]
+
+  }
+
+  object MinAggregator extends ReducingAggregator {
+
+    override type Element = Number
+
+    override def zero: Number = Double.MaxValue
+
+    override def add: (Number, Number) => Number = (n1, n2) => Math.min(n1.doubleValue(), n2.doubleValue())
+
+    override def zeroType: TypingResult = Typed[Number]
+  }
+
+  object ListAggregator extends Aggregator {
```

Comment:
What about adding First, Last? that will contain only one element?

Reply:
I added it